system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "routing_adaptor.hpp"
+
+#include <memory>
+
+namespace ad_api_adaptors
+{
+
+RoutingAdaptor::RoutingAdaptor(const rclcpp::NodeOptions & options)
+: Node("routing_adaptor", options)
+{
+  sub_goal_ = create_subscription<PoseStamped>(
+    "~/input/goal", 5, std::bind(&RoutingAdaptor::on_goal, this, std::placeholders::_1));
+  sub_waypoint_ = create_subscription<PoseStamped>(
+    "~/input/waypoint", 5, std::bind(&RoutingAdaptor::on_waypoint, this, std::placeholders::_1));
+
+  const auto node = component_interface_utils::NodeAdaptor(this);
+  node.init_cli(cli_route_);
+  node.init_cli(cli_clear_);
+  route_points_ = std::make_shared<SetRoutePoints::Service::Request>();
+}
+
+void RoutingAdaptor::on_goal(const PoseStamped::ConstSharedPtr pose)
+{
+  route_points_->header = pose->header;
+  route_points_->goal = pose->pose;
+  route_points_->waypoints.clear();
+
+  cli_clear_->async_send_request(std::make_shared<ClearRoute::Service::Request>());
+  cli_route_->async_send_request(route_points_);
+}
+
+void RoutingAdaptor::on_waypoint(const PoseStamped::ConstSharedPtr pose)
+{
+  if (route_points_->header.frame_id != pose->header.frame_id) {
+    RCLCPP_ERROR_STREAM(get_logger(), "The waypoint frame does not match the goal.");
```

Comment:
- If it can continue working, I think `WARN` is better.
- Don't we have to transform the pose in this case?

```suggestion
    RCLCPP_WARN_STREAM(get_logger(), "The waypoint frame does not match the goal.");
```

Reply:
Added `return` in 11acf52f44228d81b7399e61de90fc819aec110d.
I think it's acceptable for now.

Reply:
It cannot continue. Fixed https://github.com/autowarefoundation/autoware.universe/pull/1494/commits/11acf52f44228d81b7399e61de90fc819aec110d.